service/lib/agama/dbus/software/manager.rb

```diff
@@ -54,20 +54,20 @@ def initialize(backend, logger)
           @selected_patterns = {}
         end
 
+        # List of issues, see {DBus::Interfaces::Issues}
+        #
+        # @return [Array<Agama::Issue>]
+        def issues
+          backend.issues
+        end
+
         SOFTWARE_INTERFACE = "org.opensuse.Agama.Software1"
         private_constant :SOFTWARE_INTERFACE
 
         dbus_interface SOFTWARE_INTERFACE do
-          dbus_reader :available_base_products, "a(ssa{sv})"
-
-          dbus_reader :selected_base_product, "s"
-
-          # documented way to be able to write to patterns and trigger signal
-          attr_writer :selected_patterns
+          dbus_reader :available_products, "a(ssa{sv})"
 
-          # selected patterns is hash with pattern name as id and 0 for user selected and
-          # 1 for auto selected. Can be extended in future e.g. for mandatory patterns
-          dbus_attr_reader :selected_patterns, "a{sy}"
+          dbus_reader :selected_product, "s"
```

Comment:
when we touch it we probably also need to solve issue with version...as there can be two products with same name and different version like it happen with micro based on some SP. And registration need version....maybe product should be tuple of strings with name and version?